Description: Skip kexec_file_load test on ppc* before 4.11.
Author: Adam Conrad <adconrad@ubuntu.com>
Last-Update: 2017-06-17

--- strace-4.15.orig/tests/kexec_file_load.test
+++ strace-4.15/tests/kexec_file_load.test
@@ -3,4 +3,12 @@
 # Check decoding of kexec_file_load syscall.
 
 . "${srcdir=.}/init.sh"
+
+case $(uname -m) in
+  ppc*)
+    # kexec_file_load was wired up in 80f60e509a03ff9ff2bdbf9cd1e935c6360b8bd9
+    require_min_kernel_version_or_skip 4.11
+    ;;
+esac
+
 run_strace_match_diff -s9
